### PR TITLE
[DESKTOP] Cleaning up

### PR DESCRIPTION
**Pull request by jvnm-dev**
Things that have changed:
  - Components connected to the store => Containers
  - Cleaning up the imports: first vendors, space, ours
  - Adding FIXME to fix the things that need to be fixed
    but not now because I don't want to ;)
  - Adding missing types
  - ...

I prepared the jwt decoding process too

### CODE REVIEW

File: projects/desktop/src/Router.tsx [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
+import { RegisterContainer } from './containers/RegisterContainer'
+import { isAuthenticated } from './helpers/auth'
+import { error } from './helpers/logger'
+
+const notAuthenticated = (): JSX.Element => {
+  error('You are not authenticated!')
+  return <Redirect to='/' />
+}
+
+const PrivateRoute = ({ component, ...options }: RouteProps) => {
```
Comment (Sylv11) [inline projects/desktop/src/Router.tsx]: Possibility to provide a returned type?

File: projects/desktop/src/Router.tsx [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+  const finalComponent = isAuthenticated() ? component : notAuthenticated
+  return <Route {...options} component={finalComponent} />
+}
+
+const OnboardingTemporary = () => {
```
Comment (Sylv11) [inline projects/desktop/src/Router.tsx:20]: Do not forget to provide a type later

File: projects/desktop/src/components/common/SingleForm/checkbox/CheckboxInput.tsx [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
 
 export const CheckboxInput = styled.input`
     position: absolute;
     opacity: 0;
     cursor: pointer;
     height: 0;
     width: 0;
-    
+
     :checked ~ span {
```
Comment (Sylv11) [inline projects/desktop/src/components/common/SingleForm/checkbox/CheckboxInput.tsx:12]: Can you remember me this selector?

File: projects/desktop/src/components/common/SingleForm/input/InputsRow.tsx [hunk 4]
```diff
@@ -2,12 +2,12 @@ import styled from 'styled-components'
 
 export const InputsRow = styled.div`
   display: flex;
-  
+
   > * {
```
Comment (Sylv11) [inline projects/desktop/src/components/common/SingleForm/input/InputsRow.tsx:6]: And this selector?

File: projects/desktop/src/containers/LoginContainer.tsx [hunk 5]
```diff
@@ -1,67 +1,71 @@
...
+    email: string,
+    password: string,
+}
+
+export const LoginContainer = connect()((): JSX.Element => {
+    const {t}: UseTranslationResponse = useTranslation()
+
+    const getTitleContent = (): IDangerousHTMLContent => {
+        return {__html: t('login.title')};
+    }
```
Comment (Sylv11) [inline projects/desktop/src/containers/LoginContainer.tsx]: `const getTitleContent = (): IDangerousHTMLContent => {__html: t('login.title')}` 
Not possible?

File: projects/desktop/src/containers/RegisterContainer.tsx [hunk 6]
```diff
@@ -1,105 +1,109 @@
...
+    }
+}
+
+export const RegisterContainer = connect()((): JSX.Element => {
+    const [state, dispatch] = useReducer(reducer, initialState)
+    const {t}: UseTranslationResponse = useTranslation()
+
+    const getTitleContent = (): IDangerousHTMLContent => {
+        return {__html: t('register.title')};
+    }
```
Comment (Sylv11) [inline projects/desktop/src/containers/RegisterContainer.tsx]: ` const getTitleContent = (): IDangerousHTMLContent => {__html: t('register.title')}`
Same

File: projects/desktop/src/containers/LoginContainer.tsx [hunk 7]
```diff
@@ -1,67 +1,71 @@
...
+            <Box>
+                <LeftSide>
+                    <Logo src={logo} alt='Bitsky' />
+                    <LoginLanguageChooser />
+                    <BigTitle dangerouslySetInnerHTML={getTitleContent()} />
+
+                    <FinalForm
+                        onSubmit={onSubmit}
+                        initialValues={{ remember: false }}
+                        render={({handleSubmit}) => <Form handleSubmit={handleSubmit} />}
```
Comment (Sylv11) [inline projects/desktop/src/containers/LoginContainer.tsx]: A type for this handleSubmit?

File: projects/desktop/src/containers/RegisterContainer.tsx [hunk 8]
```diff
@@ -1,105 +1,109 @@
...
+    return (
+        <SingleFormContainer>
+            <Box>
+                <Logo src={logo} alt='Bitsky' />
+                <RegisterLanguageChooser />
+                <BigTitle dangerouslySetInnerHTML={getTitleContent()} />
+                <FinalForm
+                    onSubmit={onSubmit}
+                    initialValues={{ termsOfUse: false }}
+                    render={({handleSubmit}) => <Form handleSubmit={handleSubmit} invalid={state.invalidForm} />}
```
Comment (Sylv11) [inline projects/desktop/src/containers/RegisterContainer.tsx]: A type for this handleSubmit?

File: projects/desktop/src/helpers/auth.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import jwtDecode from 'jwt-decode'
+
+export const isAuthenticated = (): boolean => {
+    return !!localStorage.getItem('token')
+}
```
Comment (Sylv11) [inline projects/desktop/src/helpers/auth.ts]: `export const isAuthenticated = (): boolean => !!localStorage.getItem('token')`

File: projects/desktop/src/components/common/SingleForm/checkbox/CheckboxInput.tsx (hunk 3 above)
Comment (jvnm-dev) [reply projects/desktop/src/components/common/SingleForm/checkbox/CheckboxInput.tsx:12]: Annotation 2019-11-30 093427

File: projects/desktop/src/components/common/SingleForm/input/InputsRow.tsx (hunk 4 above)
Comment (jvnm-dev) [reply projects/desktop/src/components/common/SingleForm/input/InputsRow.tsx:6]: All elements where the parent is div

**Pull request merged**